src/uct/ib/rc/base/rc_iface.h

```diff
@@ -385,9 +386,14 @@ uct_rc_fc_ctrl(uct_ep_t *ep, unsigned op, uct_rc_fc_request_t *req)
 static inline uct_rc_ep_t *uct_rc_iface_lookup_ep(uct_rc_iface_t *iface,
                                                   unsigned qp_num)
 {
+    uct_rc_ep_t *ep;
+
     ucs_assert(qp_num < UCS_BIT(UCT_IB_QPN_ORDER));
-    return iface->eps[qp_num >> UCT_RC_QP_TABLE_ORDER]
-                     [qp_num &  UCS_MASK(UCT_RC_QP_TABLE_MEMB_ORDER)];
+    ucs_spin_lock(&iface->eps_lock);
+    ep = iface->eps[qp_num >> UCT_RC_QP_TABLE_ORDER]
+                   [qp_num &  UCS_MASK(UCT_RC_QP_TABLE_MEMB_ORDER)];
+    ucs_spin_unlock(&iface->eps_lock);
```

Comment:
pls remove from lookup

Reply:
ok, removed.
force pushed due to small commit